apps/demo-app/app/components/UserDataGrid.vue

```diff
@@ -0,0 +1,179 @@
+  { key: "email", label: t("dataGrid.userTable.email") },
+  { key: "age", label: t("dataGrid.userTable.age") },
+  { key: "address", label: t("dataGrid.userTable.address"), width: "max-content" },
+  { key: "city", label: t("dataGrid.userTable.city") },
+  { key: "country", label: t("dataGrid.userTable.country"), width: "max-content" },
+  { key: "role", label: t("dataGrid.userTable.role") },
+  { key: "isAdmin", label: t("dataGrid.userTable.isAdmin"), type: "boolean", width: "max-content" },
+  { key: "id", label: "", type: "detailsButton", width: "min-content" },
+];
+
+async function fetchData() {
+  try {
+    const response = await fetch("https://dummyjson.com/users");
+    if (!response.ok) {
+      throw new Error("Failed to fetch data.");
+    }
+    const result = await response.json();
+
+    const transformedUserData: UserEntry[] = result.users.map((user: TUserFromApi) => ({
+      id: user.id,
+      avatar: user.image,
+      firstName: user.firstName,
+      lastName: user.lastName,
+      email: user.email,
+      age: user.age,
+      role: user.role,
+      isAdmin: user.role === "admin",
+      address: user.address.address,
+      city: user.address.city,
+      country: user.address.country,
+    }));
+
+    userData.value = transformedUserData;
+  } catch (err) {
+    if (err instanceof Error) {
+      error.value = err.message;
+    } else {
+      error.value = "An unknown error occurred.";
+    }
+  } finally {
+    isLoading.value = false;
+  }
+}
+
+const userCustomType = createFeature(() => ({
+  name: Symbol("User Types"),
+  typeRenderer: {
+    avatar: DataGridFeatures.createTypeRenderer({
+      cell: {
+        component: ({ modelValue }) => {
+          return h(OnyxAvatar, { fullName: "Test", src: modelValue?.toString(), size: "32px" });
+        },
+      },
+    }),
+    detailsButton: DataGridFeatures.createTypeRenderer({
+      cell: {
+        component: (props) => {
+          return h(OnyxSystemButton, {
+            label: "Show details",
+            icon: iconForward,
+            link: `#some-page-${props.row.id}`,
+            onClick: () => alert(JSON.stringify(props)),
+            style: { verticalAlign: "middle" },
+          });
+        },
+      },
+    }),
+  } satisfies TypeRenderMap<UserEntry>,
+}));
+const userPagination = DataGridFeatures.usePagination({
+  pageSize: 8,
+});
+const userHiddenColumns = DataGridFeatures.useHideColumns<UserEntry>({
+  columns: {
+    id: { enabled: false },
+    avatar: { enabled: false },
+  },
+});
+const userFiltering = DataGridFeatures.useFiltering<UserEntry>({
+  columns: {
+    id: { enabled: false },
+    isAdmin: { enabled: false },
+    avatar: { enabled: false },
+  },
+});
+const userSorting = DataGridFeatures.useSorting<UserEntry>({
+  columns: {
+    id: { enabled: false },
+    avatar: { enabled: false },
+  },
+});
+
+const userStickyColumns = DataGridFeatures.useStickyColumns<UserEntry>({
+  columns: ["id"],
+  position: "right",
+});
```

Comment:
```suggestion
const userPagination = DataGridFeatures.usePagination({
  pageSize: 8,
});
const userHiddenColumns = DataGridFeatures.useHideColumns<UserEntry>();
const userFiltering = DataGridFeatures.useFiltering<UserEntry>();
const userSorting = DataGridFeatures.useSorting<UserEntry>();

const userStickyColumns = DataGridFeatures.useStickyColumns<UserEntry>({
  columns: ["id"],
  position: "right",
});
```

- would prefer to use the default configs for simplicity
- please register the filtering feature first so the search is displayed at the top of the flyout
- please also add the resize and selection feature :) We don't need any logic for the selection but we should showcase that the data grid supports this

<img width="276" height="296" alt="image" src="https://github.com/user-attachments/assets/08f541e7-9e3d-429a-bbc6-fb1ca94600a2" />